### ISSUE TITLE
path-add-suffix doesn't implement the same behavior as its documentation says

### ISSUE THREAD

**Issue by endobson** [racket/racket#1292]
http://docs.racket-lang.org/reference/Manipulating_Paths.html?q=rename-ext#%28def._%28%28lib._racket%2Fprivate%2Fmisc..rkt%29._path-add-suffix%29%29

Similar to path-replace-suffix, but any existing suffix on path is preserved by replacing every . in the last path element with _, and then the suffix is added to the end.

```
-> (path-add-suffix "foo.bar.baz" ".z")
#<path:foo.bar_baz.z>
```

I expect this to return `#<path:foo_bar_baz.z>` because the `.` between foo and bar is in the last path element.

**Comment by mflatt** [comment]
I'll fix the docs, but see also #1307. If we go with `path-add-extension`, probably it should behave more like the documented currently suggests for `path-add-suffix`.

**Comment by mflatt** [comment]
After thinking about this more, I remember why the newly documented behavior was the implemented one. The idea is that `path-add-suffix` should avoid mapping differently names files to a same-named file. If all "."s were replaced with "_", then adding ".fail" to "a.tar.gz" and "a_tar.gz" would produce "a_tar_gz.fail" in both cases. So, I think `path-add-extension` should stick with the old behavior, after all (but not count a leading "." as an extension separator).
